feat: added feature flag docs (#203)
front-commerce/developers.front-commerce.com: source/docs/advanced/features/feature-flags.md

---
id: feature-flags
title: Feature Flags
---

> Some modules in front-commerce have feature flags that are used to enable/disable set module.

Reviewer: ```suggestion
Some modules in Front-Commerce use feature flags to enable or disable its features.
```

> ### Enabling/Disabling a feature

Reviewer: ```suggestion
## Enabling or Disabling a feature
```

Typically the feature flag is defined inside the folder of the module in a loader called `featureFlagLoader.js`. So for example the feature flag for the wishlist module on magento1 is located at `src/server/modules/magento1/wishlist/featureFlagLoader.js`.

> To enable/disable a feature it is sufficient to update/override the `featureFlagLoader.js` file to return `true` (to enable the feature) or `false` (to disable the feature) when the name matches the module's name. For example setting the below in `src/server/modules/magento1/wishlist/featureFlagLoader.js` will disable the wishlist feature

Reviewer: This is not really helpful for integrators IMO.\
They should have to override it from one of their application module.

This Feature Flag custom loader is implemented a bit differently than usual in the sense that it overrides the default implementation.\
What should also be documented to make it clearer is (I think) the `contextEnhancer` too.

Author: updated it and made the `contextEnhancer` the main focus

---
id: feature-flags
title: Feature Flags
---

Some modules in Front-Commerce use feature flags to enable or disable its features.

## Enabling or Disabling a feature

> Feature flags are resolved by the [FeatureFlagLoader](https://gitlab.com/front-commerce/front-commerce/-/blob/main/src/server/modules/front-commerce/core/loaders.js#L7) which is typically overrided in the contextEnhancer of each module supporting a feature flag (e.g. [wishlist](https://gitlab.com/front-commerce/front-commerce/-/blob/main/src/server/modules/magento2/wishlist/index.js#L21), [search](https://gitlab.com/front-commerce/front-commerce/-/blob/main/src/server/modules/front-commerce/search/index.js#L41)).

Reviewer: ```suggestion
Feature flags are resolved by the [FeatureFlagLoader](https://gitlab.com/front-commerce/front-commerce/-/blob/main/src/server/modules/front-commerce/core/loaders.js#L7) which is typically overrided in the [`contextEnhancer`](/docs/reference/graphql-module-definition.html#contextEnhancer-optional) of each module supporting a feature flag (e.g. [wishlist](https://gitlab.com/front-commerce/front-commerce/-/blob/main/src/server/modules/magento2/wishlist/index.js#L21), [search](https://gitlab.com/front-commerce/front-commerce/-/blob/main/src/server/modules/front-commerce/search/index.js#L41)).
```

Reviewer: Also, links should point to a specific version (e.g: tag or commit hash) instead of branch. Otherwise, they'll quickly get out of sync and become useless.

> To enable or disable a feature it is sufficient to update the overrided `FeatureFlag` in the `contexEnhancer` of the module and make it return `true` (to enable the feature) or `false` (to disable the feature) ONLY when the name matches the module's name. For example setting the below in `src/server/modules/magento1/wishlist/featureFlagLoader.js` (used in the wishlist context enhancer) will disable the wishlist feature.

Reviewer: ```suggestion
To enable or disable a feature one must update the overriden `FeatureFlag` loader in the `contexEnhancer` of the module and ensure `isActive` returns `true` (to enable the feature) or `false` (to disable the feature) ONLY when the name matches the feature flag's name. For example setting the below in `src/server/modules/magento1/wishlist/featureFlagLoader.js` (used in the wishlist context enhancer) will disable the wishlist feature.
```
